### PR TITLE
fix obs_prep job;

### PR DESCRIPTION
**Pull request by jkuang-exploro**
establish new module load mechanism
replace bash shell with korn shell (please test on Hera)
1st batches of changes of comprehensive workflow upgrade

This PR re-enabled the obs_prep job to run on Orion, and keep Hera run unaffected.
Korn shell is used to replace bash shell in various levels of scripts related to obs_prep job. Such changes will be populated to the whole workflow if it passes test.

## Testing

Run 3dvar case on Both Orion and Hera.

### CODE REVIEW

File: jobs/JJOB_OBS_PREP [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
 
-module purge
-source $MOD_PATH/godas.main
-source $MOD_PATH/godas.python
-module list
+. $ROOT_GODAS_DIR/ush/load_godas_modules.sh
+status=$?
+[[ $status -ne 0 ]] && exit $status
+
+export OPT=/home/rmahajan/opt
```
Comment (flampouris) [inline jobs/JJOB_OBS_PREP]: We need the export OPT=/home/rmahajan/opt at a higher level not in the job scripts

File: modulefiles/orion.intel19 [hunk 2]
```diff
@@ -2,10 +2,9 @@
 ##      GODAS prerequisites
 ##
 
-export OPT=/home/rmahajan/opt
```
Comment (flampouris) [inline modulefiles/orion.intel19:5]: Why do we delete this line?

File: scripts/main_obs_prep.sh [hunk 3]
```diff
@@ -1,8 +1,10 @@
-#!/bin/bash -l
+#!/bin/ksh
```
Comment (flampouris) [inline scripts/main_obs_prep.sh:1]: without the -l it wont work on Hera

**Review by flampouris — CHANGES_REQUESTED** [review]
@JianKuang-NOAA thank you for this update.

1. This branch is behind the develop
2. There are comments through the code. 
3. Please wait to update the setup, there will be a few changes on the loaded modules.

File: scripts/main_obs_prep.sh (hunk 3 above)
Comment (flampouris) [reply scripts/main_obs_prep.sh:1]: > Modules with sh, bash, and ksh scripts
> Due to the way the POSIX standard is defined for bash, sh, and ksh you MUST add the -l option (that is a lowercase L) to the shebang (e.g. #!/bin/sh) line at the top of your script for all sh, bash, or ksh batch scripts. For example:
> 
> #!/bin/ksh -l
> 
> module load intel
> module load impi
> 
> srun -n 12 ​./xhpl


I think that the solution of "sh script.sh" and keeping the -l on the top of the script will work.

**Comment by jkuang-exploro** [comment]
@flampouris OK, I will add it.

File: jobs/JJOB_FCST_PREP [hunk 4]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash -l
+#!/bin/ksh
```
Comment (flampouris) [inline jobs/JJOB_FCST_PREP:1]: @JianKuang-NOAA we cannot remove the -l or -login, it stops working on Hera.

File: scripts/main_obs_prep.sh (hunk 3 above)
Comment (jkuang-exploro) [reply scripts/main_obs_prep.sh:1]: @flampouris adding -l to ksh won't work at all.

File: jobs/JJOB_OBS_PREP (hunk 1 above)
Comment (jkuang-exploro) [reply jobs/JJOB_OBS_PREP]: I will move it to config.base.

File: modulefiles/orion.intel19 (hunk 2 above)
Comment (jkuang-exploro) [reply modulefiles/orion.intel19:5]: @flampouris /modulefile is the wrong place to put this kind of variables.

**Comment by jkuang-exploro** [comment]
will reopen a new one.

**Pull request closed without merging**